src/context/store.ts

```diff
@@ -1,15 +1,67 @@
-import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit';
+import { Map } from 'immutable';
+import {
+  configureStore,
+  getDefaultMiddleware,
+  combineReducers,
+} from '@reduxjs/toolkit';
+import {
+  innerReducer,
+  outerReducer,
+  middleware,
+} from 'redux-async-initial-state';
+
 import mapStateReduce from './mapStateSlice';
+import serverStateReduce from './serverStateSlice';
+import { getLayersAvailableDates } from '../utils/server-utils';
+import { LAYERS_AVAILABLE_DATES } from '../constants';
+import { AvailableDates } from '../config/types';
 
-export const store = configureStore({
-  reducer: {
+const initializeStore = async (getCurrentState: any) => {
+  const isStorageSet = localStorage.getItem(LAYERS_AVAILABLE_DATES) !== null;
+
+  if (isStorageSet) {
+    // Aborting the initialization
+    return Promise.reject();
+  }
+
+  const layersAvailableDates = await getLayersAvailableDates();
+  const currentState = getCurrentState();
+  const { serverState } = currentState;
+
+  localStorage.setItem(
+    LAYERS_AVAILABLE_DATES,
+    JSON.stringify(layersAvailableDates),
+  );
+
+  return new Promise(resolve => {
+    resolve({
+      ...currentState,
+      serverState: serverState.set(
+        'availableDates',
+        Map(layersAvailableDates) as AvailableDates,
+      ),
+    });
+  });
+};
+
+const reducer = outerReducer(
+  combineReducers({
     mapState: mapStateReduce,
-  },
+    serverState: serverStateReduce,
+    asyncInitialState: innerReducer,
+  }),
+);
+
+export const store = configureStore({
+  reducer,
   // TODO: Instead of snoozing this check, we might want to
   // serialize the state
-  middleware: getDefaultMiddleware({
-    serializableCheck: false,
-  }),
+  middleware: [
+    middleware(initializeStore),
+    ...getDefaultMiddleware({
+      serializableCheck: false,
+    }),
+  ],

```

Comment:
Is this just boilerplate from async redux? It seems complicated, but I haven't used that package before.

Reply:
Yes, it was the easiest one I found ([link](https://github.com/KELiON/redux-async-initial-state))